Sources/WebSocketKit/WebSocket.swift

```diff
@@ -252,12 +269,27 @@ public final class WebSocket {
 
         // if this frame was final and we have a non-nil frame sequence,
         // output it to the websocket and clear storage
-        if let frameSequence = self.frameSequence, frame.fin {
+        if var frameSequence = self.frameSequence, frame.fin {
             switch frameSequence.type {
             case .binary:
-                self.onBinaryCallback(self, frameSequence.binaryBuffer)
+                if decompressor != nil {
```

Comment:
`if let decompressor`?

Reply:
We still need to support swifts older than 5.7.

Other than that, iirc we can't trigger a compiler copy of the decompressor.

Reply:
> We still need to support swifts older than 5.7.

Oops, I forgot about that. 😆

> Other than that, iirc we can't trigger a compiler copy of the decompressor.

Hmm interesting. Then should `Decompressor` be a class since it manages a resource?

Reply:
Then, we will have ARC overhead, which we currently don't have.
This is also not part of the API, so we don't really need to worry about it being friendly to developers, too much.
Swift NIO also had it as a struct, not class.